android_mobile/src/main/java/com/alexstyl/specialdates/events/peopleevents/AndroidPeopleEventsProvider.kt

```diff
@@ -0,0 +1,257 @@
+                queryCustomEvent(deviceEventIdFrom.get())
+            } else StandardEventType.OTHER
+        }
+        return StandardEventType.fromId(rawEventType)
+    }
+
+    @Throws(ContactNotFoundException::class)
+    private fun getContactEventFrom(cursor: Cursor, contact: Contact): ContactEvent {
+        val date = getDateFrom(cursor)
+        val eventType = getEventType(cursor)
+
+        val eventId = getDeviceEventIdFrom(cursor)
+        return ContactEvent(eventId, eventType, date, contact)
+    }
+
+    @ContactSource
+    private fun getContactSourceFrom(cursor: Cursor): Int {
+        val sourceTypeIdex = cursor.getColumnIndexOrThrow(AnnualEventsContract.SOURCE)
+        return cursor.getInt(sourceTypeIdex)
+    }
+
+    private fun queryCustomEvent(deviceId: Long): EventType {
+        return customEventProvider.getEventWithId(deviceId)
+    }
+
+    companion object {
+
+        private const val DATE_FROM = "substr(" + AnnualEventsContract.DATE + ",-5) >= ?"
+        private const val DATE_TO = "substr(" + AnnualEventsContract.DATE + ",-5) <= ?"
+        private const val DATE_BETWEEN_IGNORING_YEAR = DATE_FROM + " AND " + DATE_TO + " AND " + AnnualEventsContract.VISIBLE + " == 1"
+        private val PEOPLE_PROJECTION = arrayOf(AnnualEventsContract.DATE)
+        //    private static final Uri PEOPLE_EVENTS = PeopleEventsContract.PeopleEvents.CONTENT_URI;
+        private val PROJECTION = arrayOf(AnnualEventsContract.CONTACT_ID, AnnualEventsContract.DEVICE_EVENT_ID, AnnualEventsContract.DATE, AnnualEventsContract.EVENT_TYPE, AnnualEventsContract.SOURCE)
+
+        /*
+        We use this column in order to be able to do comparisons of dates, without having to worry about the year
+        So, instead of a full date 1990-12-19, this will return 12-19. Similarwise for --12-19.
+
+        an example in use: select * from annual_events WHERE substr(date,-5) >= '03-04' ORDER BY substr(date,-5) asc LIMIT 1
+     */
+        private const val DATE_COLUMN_WITHOUT_YEAR = "substr(" + AnnualEventsContract.DATE + ", -5) "
+
+        private fun firstHalfOf(timeDuration: TimePeriod): TimePeriod {
+            return TimePeriod.between(
+                    timeDuration.startingDate,
+                    Date.endOfYear(timeDuration.startingDate.year)
+            )
+        }
+
+        private fun secondHalfOf(timeDuration: TimePeriod): TimePeriod {
+            return TimePeriod.between(
+                    Date.startOfYear(timeDuration.endingDate.year),
+                    timeDuration.endingDate
+            )
+        }
+
+        private fun getDateFrom(cursor: Cursor): Date {
+            val index = cursor.getColumnIndexOrThrow(AnnualEventsContract.DATE)
+            val rawDate = cursor.getString(index)
+            try {
+                return DateParser.INSTANCE.parse(rawDate)
+            } catch (e: DateParseException) {
+                throw DeveloperError("Invalid date stored to database. [$rawDate]")
```

Comment:
DeveloperError doesn't give much info. Find a more appropriate exception to throw